File: StringCombination.c
#include <stdio.h>
#include <string.h>

// Function to print combinations
void combination(char str[], int index, char result[], int i)
{
    if(str[i] == '\0')
    {
        result[index] = '\0';

        if(index != 0)
            printf("%s\n", result);

        return;
    }

    // Include current character
    result[index] = str[i];
    combination(str, index + 1, result, i + 1);

    // Exclude current character
    combination(str, index, result, i + 1);
}

int main()
{
    char str[100], result[100];

    printf("Enter a string: ");
    scanf("%s", str);

    printf("String combinations are:\n");

    combination(str, 0, result, 0);

    return 0;
}

File: StringCombination.java
import java.util.Scanner;

public class StringCombination {

    static void combination(String str, String result, int index) {

        if(index == str.length()) {
            if(!result.equals(""))
                System.out.println(result);
            return;
        }

        // Include character
        combination(str, result + str.charAt(index), index + 1);

        // Exclude character
        combination(str, result, index + 1);
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a string: ");
        String str = sc.nextLine();

        System.out.println("String combinations are:");

        combination(str, "", 0);
    }
}
